### ISSUE TITLE
improve performance when consent string uses range segments instead bitfields

### ISSUE THREAD

**Issue by peczenyj** [peczenyj/GDPR-IAB-TCFv2#16]
the implementation that uses range segments is correct, but slow.

```
--Parse--------------
            Rate    range bitfield
range    14355/s       --     -67%
bitfield 43840/s     205%       --
--TO_JSON------------
           Rate    range bitfield
range     272/s       --     -90%
bitfield 2595/s     853%       --
--Parse + TO_JSON----
           Rate    range bitfield
range     266/s       --     -89%
bitfield 2378/s     795%       --
```

this is more dramatic when it we are converting to JSON (we have a difference of 10x), because the implementation calls the public methods `vendor_consent` and `vendor_legitimate_interest` for all possible vendor ids between 1 and max vendor id.

an obvious solution is to parse a range section but create a bitfield. this allow us to reduce the difference between the two implementations.

a better solution is to convert the vendor list directly into a json format with just one method.

**Comment by peczenyj** [comment]
new benchmark

```
--Parse--------------
            Rate    range bitfield
range    10960/s       --     -71%
bitfield 37594/s     243%       --
--TO_JSON------------
           Rate bitfield    range
bitfield 4842/s       --     -36%
range    7592/s      57%       --
--Parse + TO_JSON----
           Rate bitfield    range
bitfield 4147/s       --      -3%
range    4267/s       3%       --
```

comparing with v0.081 we have an improvement on TO_JSON method of

range based: 3116% faster
bitfield: 76% faster

FYI the current version add a new section on the JSON (publisher restrictions).

By removing this section we find the new benchmark is even better

```
--TO_JSON------------
           Rate bitfield    range
bitfield 4931/s       --     -37%
range    7790/s      58%       --
```

**Comment by peczenyj** [comment]
fixed on v0.082
